joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -325,6 +360,33 @@ JointTrajectoryController::halt()
   set_op_mode(hardware_interface::OperationMode::ACTIVE);
 }
 
+void JointTrajectoryController::publish_state()
+{
+  if (state_publisher_period_.seconds() > 0.0 &&
```

Comment:
can this ever happen when being check during initialization?

Reply:
Also again only style, but I would also opt for early return here to avoid the nested `if`s

Reply:
yes, giving negative/zero state_publish_rate resolves its results to state_publisher_period_ being 0.
f668d37